upstream-sha256: 00d1272d1df1cd18e07f92e58ff770430b9b519c1ce0af052b36e3979726e7f7
actual-sha256: 150a3da0060d99f717c46b9570a8d32c1e55b9b0c102ff0256edf3e3c485890b
--- upstream.expected
+++ tolk.actual
@@ -17,8 +17,9 @@
       *(buf0+0) = ((*(buf0+0))+val0);
     }
     float val1 = (*(buf0+0));
-    float alu4 = ((val1<0.0f)?0.0f:val1);
-    *(buf1+0) = ((*(buf1+0))+alu4);
+    float alu4 = ((val1!=val1)?val1:0.0f);
+    float alu5 = ((0.0f<val1)?val1:alu4);
+    *(buf1+0) = ((*(buf1+0))+alu5);
   }
   *(buf3+lidx0) = (*(buf1+0));
   __builtin_amdgcn_fence(__ATOMIC_RELEASE, "workgroup");__builtin_amdgcn_s_barrier();__builtin_amdgcn_fence(__ATOMIC_ACQUIRE, "workgroup");
